python/pyspark/sql/classic/dataframe.py

```diff
@@ -1699,6 +1700,19 @@ def transform(
         ), "Func returned an instance of type [%s], " "should have been DataFrame." % type(result)
         return result
 
+    def pipe(
+        self, *funcs: tuple[Callable[..., ParentDataFrame]]
+    ) -> ParentDataFrame:
+        result = functools.reduce(
+            lambda init, func: init.transform(func),
```

Comment:
I would't add this API - it can be easily done at user side as you're demonstrating here.

Reply:
I feel like the same could be said about `DataFrame.transform`.

Before this was in the API, users would have to write `h(g(f(input_df)))`. With transform, users now write `input_df.transform(f).transform(g).transform(h)`. Implementing `DataFrame.pipe` would shorten this expression to `input_df.pipe(f, g, h)`.

Reply:
@HyukjinKwon any thoughts on my comment above?